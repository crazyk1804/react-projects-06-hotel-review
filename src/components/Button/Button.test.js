import React from 'react';
// import ShallowRenderer from 'react-test-renderer/shallow';
import Button from "./Button";
import {shallow} from "enzyme";

describe('the <Button/> component', () => {
	// const renderer = new ShallowRenderer();


	// it('should render', () => {
	// 	const children = 'This is a button';
	// 	renderer.render(<Button>{ children}</Button>);
	// 	const result = renderer.getRenderOutput();
	//
	// 	expect(result).toMatchSnapshot();
	// });
	//
	it('should render the correct children', () => {
		const children = 'This is a button';
		// renderer.render(<Button>{children}</Button>);
		// const component = renderer.getRenderOutput();
		const component = shallow(<Button>{children}</Button>);
		expect(component.props().children).toEqual(children);
	});

	it('should handle the onClick event', () => {
		const mockOnClick = jest.fn();
		const component = shallow(<Button onClick={mockOnClick}/>);

		component.simulate('click');
		expect(mockOnClick).toHaveBeenCalled();
	})
})