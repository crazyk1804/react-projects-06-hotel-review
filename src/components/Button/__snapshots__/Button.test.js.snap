// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <Button/> component should render 1`] = `
<ForwardRef(styled.button)>
  This is a button
</ForwardRef(styled.button)>
`;

exports[`the <Button/> component with shallow of the enzyme should render 1`] = `ShallowWrapper {}`;
